(motion_vectorization) root@30fb68955ee8:/workspace/MotionVectorization# # إنشاء مجلد logs
mkdir -p motion_vectorization/logs

# إصلاح مشكلة scipy
conda activate motion_vectorization
pip uninstall scipy -y
pip install scipy==1.10.1

# تعديل سكريبت extract_flow
sed -i 's|python3 -m RAFT.extract_flow|cd /workspace/MotionVectorization && python3 RAFT/extract_flow.py|g' scripts/extract_flow.sh

# اختبار المشروع
./scripts/script.sh videos/test.txt
Found existing installation: scipy 1.8.1
Uninstalling scipy-1.8.1:
  Successfully uninstalled scipy-1.8.1
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager, possibly rendering your system unusable.It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv. Use the --root-user-action option if you know what you are doing and want to suppress this warning.
Collecting scipy==1.10.1
  Downloading scipy-1.10.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (58 kB)
Requirement already satisfied: numpy<1.27.0,>=1.19.5 in /opt/miniconda3/envs/motion_vectorization/lib/python3.8/site-packages (from scipy==1.10.1) (1.22.3)
Downloading scipy-1.10.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (34.5 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 34.5/34.5 MB 124.0 MB/s eta 0:00:00
Installing collected packages: scipy
Successfully installed scipy-1.10.1
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager, possibly rendering your system unusable.It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv. Use the --root-user-action option if you know what you are doing and want to suppress this warning.
videos/test.txt
PREPROCESS
test1
RAFT
test1
./scripts/extract_flow.sh: line 14: cd: too many arguments
videos/test.txt
CLUSTER
test1
Traceback (most recent call last):
  File "/opt/miniconda3/envs/motion_vectorization/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/miniconda3/envs/motion_vectorization/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/workspace/MotionVectorization/motion_vectorization/extract_clusters.py", line 14, in <module>
    from .utils import *
  File "/workspace/MotionVectorization/motion_vectorization/utils.py", line 14, in <module>
    from torchvision.transforms import Resize
ModuleNotFoundError: No module named 'torchvision'
TRACK
Logging to motion_vectorization/logs/test1_track_15071.out
Traceback (most recent call last):
  File "/opt/miniconda3/envs/motion_vectorization/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/miniconda3/envs/motion_vectorization/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/workspace/MotionVectorization/motion_vectorization/extract_shapes.py", line 19, in <module>
    from .dataloader import DataLoader
  File "/workspace/MotionVectorization/motion_vectorization/dataloader.py", line 7, in <module>
    from .utils import get_numbers
  File "/workspace/MotionVectorization/motion_vectorization/utils.py", line 14, in <module>
    from torchvision.transforms import Resize
ModuleNotFoundError: No module named 'torchvision'
OPTIM
Logging to motion_vectorization/logs/test1_optim_22950.out
Traceback (most recent call last):
  File "/opt/miniconda3/envs/motion_vectorization/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/miniconda3/envs/motion_vectorization/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/workspace/MotionVectorization/motion_vectorization/optimize_shapes.py", line 21, in <module>
    from .utils import *
  File "/workspace/MotionVectorization/motion_vectorization/utils.py", line 14, in <module>
    from torchvision.transforms import Resize
ModuleNotFoundError: No module named 'torchvision'
PROGRAM
Traceback (most recent call last):
  File "/opt/miniconda3/envs/motion_vectorization/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/miniconda3/envs/motion_vectorization/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/workspace/MotionVectorization/motion_vectorization/create_motion_file.py", line 12, in <module>
    from .utils import get_numbers
  File "/workspace/MotionVectorization/motion_vectorization/utils.py", line 14, in <module>
    from torchvision.transforms import Resize
ModuleNotFoundError: No module named 'torchvision'
(motion_vectorization) root@30fb68955ee8:/workspace/MotionVectorization# 